platform/genode/cai-block-client.adb

```diff
@@ -42,96 +51,49 @@ package body Cai.Block.Client is
       Cxx.Block.Client.Finalize (C.Instance);
    end Finalize;
 
-   function Convert_Request (R : Request) return Cxx.Block.Request.Class
-   is
-      Cr : Cxx.Block.Request.Class := Cxx.Block.Request.Class' (
-         Kind => Cxx.Block.None,
-         Uid => Cxx.Unsigned_Char_Array (R.Priv),
-         Start => 0,
-         Length => 0,
-         Status => Cxx.Block.Raw);
-   begin
-      case R.Kind is
-         when None =>
-            Cr.Kind := Cxx.Block.None;
-         when Sync =>
-            Cr.Kind := Cxx.Block.Sync;
-         when Read | Write =>
-            Cr.Kind := (if R.Kind = Read then Cxx.Block.Read else Cxx.Block.Write);
-            Cr.Start := Cxx.Genode.Uint64_T (R.Start);
-            Cr.Length := Cxx.Genode.Uint64_T (R.Length);
-            if R.Status = Raw then
-               Cr.Status := Cxx.Block.Raw;
-            end if;
-            if R.Status = Ok then
-               Cr.Status := Cxx.Block.Ok;
-            end if;
-            if R.Status = Error then
-               Cr.Status := Cxx.Block.Error;
-            end if;
-            if R.Status = Acknowledged then
-               Cr.Status := Cxx.Block.Ack;
-            end if;
-      end case;
-      return Cr;
-   end Convert_Request;
-
-   function Convert_Request (CR : Cxx.Block.Request.Class) return Request
-   is
-      R : Request ((case CR.Kind is
-                     when Cxx.Block.None => None,
-                     when Cxx.Block.Sync => Sync,
-                     when Cxx.Block.Read => Read,
-                     when Cxx.Block.Write => Write));
-   begin
-      R.Priv := Private_Data (CR.Uid);
-      case R.Kind is
-         when None | Sync =>
-            null;
-         when Read | Write =>
-            R.Start := Id (CR.Start);
-            R.Length := Count (CR.Length);
-            R.Status :=
-               (case CR.Status is
-                  when Cxx.Block.Raw => Raw,
-                  when Cxx.Block.Ok => Ok,
-                  when Cxx.Block.Error => Error,
-                  when Cxx.Block.Ack => Acknowledged);
-      end case;
-      return R;
-   end Convert_Request;
-
    function Ready (C : Client_Session; R : Request) return Boolean
    is
    begin
-      return Cxx.Block.Client.Ready (C.Instance, Convert_Request (R)) = 1;
+      return Cxx.Block.Client.Ready (C.Instance, Util.Convert_Request (R)) = 1;
    end Ready;
 
+   function Supported (C : Client_Session; R : Request) return Boolean
+   is
+   begin
+      return Cxx.Block.Client.Ready (C.Instance, Util.Convert_Request (R)) = 1;
```

Comment:
Shouldn't this call `Cxx.Block.Client.Supported`?

Reply:
I fixed this in https://github.com/Componolit/ada-interface/pull/28/commits/99692e51f5b5ee13f94569391b4db6f23a9cbf87.